ios/browser/api/brave_shields/adblock_engine.mm

```diff
@@ -0,0 +1,134 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/ios/browser/api/brave_shields/adblock_engine.h"
+
+#include "base/strings/sys_string_conversions.h"
+#include "brave/base/mac/conversions.h"
+#include "brave/components/adblock_rust_ffi/src/wrapper.h"
+#include "brave/components/brave_shields/common/adblock_domain_resolver.h"
+
+#if !defined(__has_feature) || !__has_feature(objc_arc)
+#error "This file requires ARC support."
+#endif
+
+@interface AdblockEngineMatchResult ()
+@property(nonatomic, readwrite) bool didMatchRule;
+@property(nonatomic, readwrite) bool didMatchException;
+@property(nonatomic, readwrite) bool didMatchImportant;
+@property(nonatomic, readwrite, copy) NSString* redirect;
+@end
+
+@implementation AdblockEngineMatchResult
```

Comment:
can you add thread checks to make sure this class is always called on the same thread/sequence? The adblock engine is not thread safe

Reply:
As mentioned in DM, iOS side has the adblock engine wrapped in a serial GCD queues which is incompatible with Chromium's sequenced thread checker.  Would be better to get this part added to the `adblock::Engine` class itself and then remove the individual thread-safety implementations in both brave-core and brave-ios